Detectors/TRD/reconstruction/src/CruRawReader.cxx

```diff
@@ -260,19 +353,30 @@ int CruRawReader::parseDigitHCHeader(int hcid)
   if (mDigitHCHeader.major == 0 && mDigitHCHeader.minor == 0 && mDigitHCHeader.numberHCW == 0) {
     //hack this data into something resembling usable.
     mDigitHCHeader.major = mHalfChamberMajor;
-    mDigitHCHeader.minor = 42; // to keep me entertained
+    mDigitHCHeader.minor = 42;
     mDigitHCHeader.numberHCW = mHalfChamberWords;
+    if (mHalfChamberWords == 0 || mHalfChamberMajor == 0) {
+      if (mMaxWarnPrinted > 0) {
+        LOG(warn) << "halfchamber header is corrupted and you have only set the halfchamber command line option to zero, hex dump of data and revisit what it should be.";
```

Comment:
where is the hex dump?

Reply:
no hex dump, you are going to have to take a hex dump of a timeframe and work through it to figure out what the data is.